templates/javadc.json

```diff
@@ -0,0 +1,214 @@
+                              "name": "SPARK_OPTIONS",
+                              "value": "${SPARK_OPTIONS}"
+                           },
+                           {
+                              "name": "APP_MAIN_CLASS",
+                              "value": "${APP_MAIN_CLASS}"
+                           },
+                           {
+                             "name": "OSHINKO_DEL_CLUSTER",
+                             "value": "${OSHINKO_DEL_CLUSTER}"
+                           },
+                           {
+                              "name": "APP_EXIT",
+                              "value": "${APP_EXIT}"
+                           },
+                           {
+                              "name": "OSHINKO_NAMED_CONFIG",
+                              "value": "${OSHINKO_NAMED_CONFIG}"
+                           },
+                           {
+                              "name": "OSHINKO_SPARK_DRIVER_CONFIG",
+                              "value": "${OSHINKO_SPARK_DRIVER_CONFIG}"
+                           },
+                           {
+                              "name": "POD_NAME",
+                              "valueFrom":
+                                {
+                                   "fieldRef":
+                                      {
+                                         "fieldPath": "metadata.name"
+                                      }
+                                }
+                           }
+                        ],
+                        "resources": {},
+                        "terminationMessagePath": "/dev/termination-log",
+                        "imagePullPolicy": "IfNotPresent",
+                        "volumeMounts": [
+                           {
+                              "mountPath": "/etc/podinfo",
+                              "name": "podinfo",
+                              "readOnly": false
+                           }
+                        ]
+                     }
+                  ],
+                  "volumes": [
+                     {
+                        "downwardAPI": {
+                           "items": [
+                              {
+                                 "fieldRef": {
+                                    "fieldPath": "metadata.labels"
+                                 },
+                                 "path": "labels"
+                              }
+                           ]
+                        },
+                        "name": "podinfo"
+                     }
+                  ],
+                  "restartPolicy": "Always",
+                  "serviceAccount": "oshinko",
+                  "dnsPolicy": "ClusterFirst"
+               }
+            }
+         }
+      },
+      {
+         "kind": "Service",
+         "apiVersion": "v1",
+         "metadata": {
+            "name": "${APPLICATION_NAME}",
+            "labels": {
+                "app": "${APPLICATION_NAME}"
```

Comment:
We need to take a closer look at including "app" labels. I discovered I believe that the "app" label is a product of "oc new-app" versus for instance "oc create -f | oc process -"

That being the case, we shouldn't add an app label ourselves, and we shouldn't count on them being present.